docs/user-guide/arrays.rst

```diff
@@ -30,6 +30,29 @@ module documentation.
 
 .. _user-guide-array:
 
+Initializing with Data
+----------------------
+Pass existing data during array creation for better performance:
```

Comment:
I don't think we should emphasize the performance benefit here, as we do not have measurements to support this claim. I would instead direct people to use the `data` parameter because it is convenient.